src/main.py

```diff
@@ -0,0 +1,80 @@
+import argparse
+
+from src.config import is_invalid, is_sharing_enabled, load_config, replace_placeholder
+from src.rds import (
+    copy_snapshot,
+    does_target_exists,
+    init_client,
+    restore_snapshot,
+    share_snapshot,
+)
+from src.tf import get_outputs
+from src.utils import assume_aws_role, setup_custom_logger
+
+LOGGER = setup_custom_logger("root")
+
+
+def main(command_line=None):
+    parser = argparse.ArgumentParser(description="Restore RDS snapshot")
+    parser.add_argument("-c", "--config", required=True)
+    parser.add_argument("-t", "--tfstate")
+    args = parser.parse_args(command_line)
+
+    data = load_config(args.config)
+    status = is_invalid(data)
+    if status:
+        LOGGER.error(status)
+        exit(1)
+
+    source = data["Source"]
+    target = data["Target"]
+    target_credentials = assume_aws_role(target["AssumeRole"], "target")
+    source_share_credentials = assume_aws_role(source["Share"]["AssumeRole"], "source")
+
+    tf_outputs = get_outputs(target_credentials, args.tfstate)
+    if tf_outputs is None:
+        LOGGER.error("TF state file does not exists")
+        exit(1)
+
+    source = replace_placeholder(source, tf_outputs, target_credentials)
```

Comment:
should the last argument here be source_share_credentials?

Reply:
The config, if is in SSM will be stored in the target account as here the DB will get created from the snapshot.
The source_share_credentials is optional and only needed when the source DB is located in a different AWS account.